### ISSUE TITLE
Login redirects to forbidden or not found errors under some circumstances in Firefox

### ISSUE THREAD

**Issue by garymoon** [go-gitea/gitea#24549]
### Description

In Firefox, when logged in as a user with some level of privilege (see a private repo, admin etc), if you log out on a page that requires that privilege to view, and then log back in as a user without that privilege, you will be redirected to either a 404 or 403 depending on the nature of the access required.

Ideally there might be a way for Gitea to handle this gracefully, but if that's more troublesome that it sounds, wrapping forbidden in the base templates so users can navigate away from it would be nice.

Thank you for Gitea! :blue_heart: 

### Gitea Version

main branch

### Can you reproduce the bug on the Gitea demo site?

Yes

### Screenshots

1. Start out as an admin, in the admin panel, and sign out
![image](https://github.com/go-gitea/gitea/assets/4531887/cdca3323-a3de-4c8a-b6d6-ab190f08b2aa)

2. Sign back in as an unprivileged user
![image](https://github.com/go-gitea/gitea/assets/4531887/bfadb043-b2ee-4ea1-8eca-e75b471b2b0c)

3. You're redirected back to the admin panel, which you don't have access to, and shown a forbidden message
![image](https://github.com/go-gitea/gitea/assets/4531887/2bf0437f-c56a-4e7e-8b46-562716daa95b)

### Operating System

Firefox

### How are you running Gitea?

`TAGS="sqlite sqlite_unlock_notify" make watch`

### Database

SQLite

**Comment by yardenshoham** [comment]
I can't reproduce. Can you provide an ordered list of steps?

**Comment by lunny** [comment]
I think it's the right response to return 404 if login user have no permission to a private repository.

**Comment by jtran** [comment]
@garymoon, what's the actual problem here?  Can you rephrase the issue by specifying both the expected/desired behavior and the current actual behavior?

**Comment by garymoon** [comment]
Oh sorry I missed the previous updates everyone, will update in the next few days :+1:

**Comment by garymoon** [comment]
Sorry for the delay all. I've outlined the steps below.

404 and 403 respectively for a repo you don't have access to and the admin panel if you don't have access make sense of course, but from a UX perspective it's jarring and sometimes confusing. The forbidden as shown in the screenshots is definitely the worse of the two, because it's not wrapped in the base template so there's no way for a user to navigate away from it. Ideally Gitea wouldn't automatically redirect a user to a page they don't have access to.

1. Start out as an admin, in the admin panel, and sign out
![image](https://github.com/go-gitea/gitea/assets/4531887/cdca3323-a3de-4c8a-b6d6-ab190f08b2aa)


2. Sign back in as an unprivileged user
![image](https://github.com/go-gitea/gitea/assets/4531887/bfadb043-b2ee-4ea1-8eca-e75b471b2b0c)


3. You're redirected back to the admin panel, which you don't have access to, and shown a forbidden message
![image](https://github.com/go-gitea/gitea/assets/4531887/2bf0437f-c56a-4e7e-8b46-562716daa95b)

**Comment by lunny** [comment]
`Sign out` should not store the previous url into cookie.

**Comment by yardenshoham** [comment]
Can you test on the latest main? I can't reproduce

1. Start out as an admin, in the admin panel, and sign out
![image](https://github.com/go-gitea/gitea/assets/20454870/f0214b60-2e37-41ce-9b62-c71509cc5edb)

2. Click sign in
![image](https://github.com/go-gitea/gitea/assets/20454870/1ad4b2cb-9e1f-47d5-ad04-6bd5d6f573a8)

3. Sign back in as an unprivileged user
![image](https://github.com/go-gitea/gitea/assets/20454870/5660ffed-b85c-4155-bbb7-a783c62ff8e4)

4. Looks good
![image](https://github.com/go-gitea/gitea/assets/20454870/5858dc30-d17f-4f4d-aee5-52290812015a)

How did you sign back in as an unprivileged user without clicking sign in?

**Comment by garymoon** [comment]
I also cannot repro this in Chrome, it appears to be a Firefox-only issue. Will update the OP accordingly.

> How did you sign back in as an unprivileged user without clicking sign in?

I'm not sure what the confusion is here, but I do click sign in.

**Comment by yardenshoham** [comment]
> I'm not sure what the confusion is here, but I do click sign in.

What I meant is from your screenshots it looked like as soon as you signed out you landed in the login page whereas I had to click sign in to get to that page.

**Comment by garymoon** [comment]
Oh right, two separate sign-in buttons :joy: In Firefox, when I sign out from the admin panel, I am redirected to the sign-in page, so I didn't need to perform that step.

**Comment by yardenshoham** [comment]
So that's what makes the difference. I wonder why firefox redirects to a different page than chrome
